Lib/test/test_logging.py

```diff
@@ -4325,18 +4326,29 @@ def test_issue_102402_msecs_has_no_floating_point_precision_loss(self):
                 patched_ns.return_value = ns
                 record = logging.makeLogRecord({'msg': 'test'})
             self.assertEqual(record.msecs, want)
+            self.assertEqual(record.created, ns / 1e9)
 
-    def test_issue_102402_relativeCreated_has_higher_precision(self):
+    def test_relativeCreated_has_higher_precision(self):
+        # See issue gh-102402
         ns = 1_677_903_920_000_998_503  # approx. 2023-03-04 04:25:20 UTC
-        offset_ns = 200
-        with patch("time.time_ns") as patched_ns:
-            patched_ns.return_value = ns
-            importlib.reload(logging)  # force logging._startTime to use mocked value
-            patched_ns.return_value = ns + offset_ns
-            record = logging.makeLogRecord({'msg': 'test'})
-        self.assertAlmostEqual(record.created, ns / 1e9, places=6)
-        # After gh-102412, precision (places) increases from 3 to 7
-        self.assertAlmostEqual(record.relativeCreated, offset_ns / 1e6, places=7)
+        for offset_ns in (200, 500, 12_354, 99_999, 1_677_903_456_999_123_456):
+            new_ns = ns + offset_ns
+            with patch("time.time_ns") as patched_ns:
+                orig_modules = import_helper._save_and_remove_modules(['logging'])
+                try:
+                    # mock for module import
+                    patched_ns.return_value = ns
+                    import logging
```

Comment:
I think that it can be out of the loop. You only need the following code in the loop:
```python
new_ns = ns + offset_ns
patched_ns.return_value = new_ns
record = logging.makeLogRecord({'msg': 'test'})
self.assertAlmostEqual(record.created, new_ns / 1e9, places=6)
self.assertAlmostEqual(record.relativeCreated, offset_ns / 1e6, places=7)
```
It is just an optimization.

Reply:
Agreed. It may be "just an optimization" but I wouldn't consider it a _premature_ optimization. Updated.